lib/mindwendel/ideas.ex

```diff
@@ -134,6 +134,53 @@ defmodule Mindwendel.Ideas do
     Lanes.broadcast_lanes_update(brainstorming_id)
   end
 
+  @doc """
+  Updates the sorting of ideas based on the order of labels occuring in filter_ids_order
+
+  ## Examples
+
+      iex> update_idea_positions_for_brainstorming_by_labels(3, [1,2,3])
+      %{1, nil}
+
+  """
+  def update_idea_positions_for_brainstorming_by_labels(brainstorming_id, filter_ids_order) do
```

Comment:
@PragTob if you have any idea how to improve the algorithm - feel free to comment 😂 but it seems to work...the idea is: If you have an active filter (blue) and already sorted your cards, e.g. blue 1, blue 2, blue 3 and you add another filter (black), then the cards shouldn't be shuffled due to old positions, but the old order of the blue cards should remain the same. This is exactly what's happening here. But yes, it turned out to be a little complex.

Reply:
What happens now is: Every time you change the filters, the existing cards are preferred in their positioning to be first. Second filter is old `position_order` to let them have their old position relative to each other.

Reply:
![Bildschirmfoto 2024-10-09 um 18 59 46](https://github.com/user-attachments/assets/7f6a7a18-32ff-4549-a2fe-fb7b7bff95cc)
![Bildschirmfoto 2024-10-09 um 18 59 53](https://github.com/user-attachments/assets/fb00ba43-6ae1-45bc-9269-6370f9f3125d)

Reply:
@JannikStreek  Oof, I think that's very confusing that blue/Cyan is suddenly first in the order so that filtering it out and back in (and maybe moving cards inbetween) changes the display order. Like, what if order matters for my school project and I'm ranking all my favorite foods and of course all the ones tagged "Burger" are first. Then I go and focus on Pizza, so I only sort those and all of the sudden Burgers are ranked below Pizzas?

Like, it's a stupid example but I can imagine something like that happen in a school easily. It's tough but I think they should only change their position relative to each other and not affect the positioning of non visible cards, which is of course tough/not easy to get right.